testing/infra/terraform/modules/soaktest_workers/main.tf

```diff
@@ -0,0 +1,141 @@
+locals {
+  executable_name    = "apmsoak"
+  service_name       = "apmsoak.service"
+  network_name       = "apmsoak-network"
+  remote_working_dir = "apmsoak/"
+  config_hash = sha256(join("_",
+    [
+      var.apm_server_url,
+      var.apm_secret_token,
+      var.apm_loadgen_max_rate,
+      var.apm_loadgen_agents_replicas,
+      var.elastic_agent_version,
+      var.fleet_url,
+    ]
+  ))
+}
+
+data "google_compute_image" "worker_image" {
+  project = "ubuntu-os-cloud"
+  family  = "ubuntu-minimal-2004-lts"
+}
+
+data "google_client_openid_userinfo" "me" {}
+
+resource "google_os_login_ssh_public_key" "cache" {
+  user = data.google_client_openid_userinfo.me.email
+  key  = file(var.public_key)
+}
+
+resource "google_compute_network" "apmsoak_worker_network" {
+  name = local.network_name
+}
+
+resource "google_service_account" "worker_svc_account" {
+  account_id   = "apmserver-soaktest-worker"
+  display_name = "APM Server soaktest worker"
+}
+
+resource "google_compute_firewall" "allow_ssh" {
+  name    = "apmserver-soaktest-allowssh"
+  network = local.network_name
+
+  depends_on = [
+    google_compute_network.apmsoak_worker_network
+  ]
+
+  allow {
+    protocol = "tcp"
+    ports    = ["22"]
+  }
+  source_ranges = ["0.0.0.0/0"]
+}
+
+resource "google_compute_instance" "worker_instance" {
+  # Trigger a recreate of workers on config changes
+  name         = substr("apmsoak-worker-${local.config_hash}", 0, 63)
+  machine_type = "e2-micro"
+  zone         = var.gcp_zone
+
+  boot_disk {
+    initialize_params {
+      image = data.google_compute_image.worker_image.self_link
+    }
+  }
+
+  network_interface {
+    network = local.network_name
+    access_config {}
+  }
+
+  depends_on = [
+    google_compute_firewall.allow_ssh,
+  ]
```

Comment:
Same here

Reply:
Didn't get this one. While provisioning we require firewall to be created however, we don't actually use the firewall.

Reply:
Oh, I see. Ignore this comment then